packages/iTwinUI-react/src/core/Table/Table.tsx

```diff
@@ -659,6 +663,10 @@ export const Table = <
     setPageSize(pageSize);
   }, [pageSize, setPageSize]);
 
+  const numSelectedRows =
+    selectionMode === 'single'
+      ? 0
```

Comment:
comment from @rohan-kadkol 

> What if one row is selected when `selectionMode` is single? Then `numSelectedRows` should be 1?

Reply:
This was an issue for me. For one, if there is single selection mode, I do not think we should show how many rows are selected since the max amount of rows that can be selected in single selection mode is 1. I set this to 0 so that the checked rows counter isn't displayed in this case. 

With that being said, it does make the variable name misleading :/

Reply:
I don't think there is a best way here. For example, it could be beneficial to show that there is already one row selected when you filtered table and that selected row is not visible or when you go to the next page, but also you should remember that you selected just one row.

Reply:
Ohh, right, that is a good point. Also if we don't want to display it when `selectionMode = single`, can we maybe pass the selectionMode to the TablePaginator so that it can only display the row count when `selectionMode === single && totalSelectedRowsCount > 0` instead of just the second condition? (on line 286 of `TablePaginator.tsx`)

Reply:
I think it's unnecessary to pass the `selectionMode` prop into `TablePaginatorRendererProps`

Reply:
What if we have a prop to allow users to not display the selected count? That way `totalSelectedRowCount` can store the real selectedCount for both `single` (0-1) and `multiple` (0+) selectionModes. Then if the user doesn't want the selected count, they could pass `false`.

Reply:
I think then let's go ahead with having that 0 there. But it would help to add a comment saying why `totalSelectedRowsCount` can be 0 even when one row is selected. Else, someone in the future might get confused that this is a bug.

Reply:
I saw your comment on [L94-96](https://github.com/iTwin/iTwinUI-react/pull/837/files#diff-9edd44b431060c27e0260d286ee188bdf901225b1ee84ff6540f9d796dd0f999R94-R96). But better to also write a comment on [L669-670](https://github.com/iTwin/iTwinUI-react/pull/837/files#diff-9edd44b431060c27e0260d286ee188bdf901225b1ee84ff6540f9d796dd0f999R669-R670). Probably something like:

```tsx
totalSelectedRowsCount:
        selectionMode === 'single' ? 0 : instance.selectedFlatRows.length, // 0 when selectionMode = 'single' since totalSelectedRowCount is for multi-selection mode only
```

Reply:
This comment has been added